### ISSUE TITLE
Issues around Ayah highlighting updating

### ISSUE THREAD

**Issue by ismail-s** [quran/quran_android#1375]
Salaams,

I've found that whilst listening to the audio, the highlight that identifies the currently playing Ayah doesn't always update perfectly/immediately. Rather than raising separate issues, I've combined them as they'd probably get fixed together (can separate them if it helps though).

# Scenario 1
- Start playing the audio with Arabic text shown, with the audio playing the last Ayah on the page
- Lock the screen
- Wait for the audio to get onto the next page
- Unlock the screen

Expected - no Ayah on the first page is highlighted
Actual - the last Ayah on the first page is highlighted until the next Ayah finishes

# Scenario 2
Do the same as scenario 1, but with the translation showing instead the whole time

Expected - no Ayah on the first page is highlighted
Actual - the last Ayah on the first page is highlighted, and stays highlighted until that fragment gets destroyed.

# Scenario 3
When switching from Arabic to translation, the highlight doesn't immediately follow (ie it takes a few seconds for the highlight to get sorted). This seems to be a more general issue with the highlight for the translation.

**Comment by ahmedre** [comment]
jazakAllah khairan for this report. for scenarios 1 and 2, i think we stop listening for updates when the activity is paused and that's probably why - that may be ok but then we should update right away as opposed to waiting for the next ayah to finish - will have to see why this is. not sure about scenario 3 but in sha' Allah can look into it.

**Comment by nacer80** [comment]
Salam 3alikom 
My brothers
The issue is when we have tow  Surahs in one page, so when we need to play the 1st one from the Ayats in the page  between tow surahs does not play so we have to download the second Surah
(As I mentioned before for ex try to keep surah Al3la number 87 and delete Elkhashi3a number 88 and try to play the Ayats from Surah Ala3la in the page between the both Surahs)

**Comment by ismail-s** [comment]
@nacer80 I agree with what you said (sorry, I didn't fully understand from your first comment, I've understood now). I've checked and I also get the same issue.

But I think your issue should be raised as a separate issue.

**Comment by ismail-s** [comment]
I thought I'd seen this behaviour partly fixed already, so I retested, and was still able to reproduce. But scenario 1 now seems to be failing in the same way as scenario 2 ie the failing behaviour seems to be more consistent between Arabic and translation.

I was hoping to try and figure out how to fix this, but the code seems quite complex around this, and split across multiple files/places, so I'm not sure I'll be able to without spending lots of time.
